nixos/modules/system/boot/systemd/user.nix

```diff
@@ -93,6 +107,42 @@ in {
       description = lib.mdDoc "Definition of systemd per-user timer units.";
     };
 
+    systemd.user.tmpfiles = {
+      rules = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        example = [ "D %C - - - 7d" ];
+        description = lib.mdDoc ''
+          User rules for creation, deletion and cleaning of volatile and
+          temporary files automatically. See
+          {manpage}`tmpfiles.d(5)`
+          for the exact format.
+        '';
+      };
+
+      users = mkOption {
+        description = mdDoc ''
+          Per-user rules for creation, deletion and cleaning of volatile and
```

Comment:
I think this per-user is a bit confusing. 

The `systemd.user.tmpfiles` option will apply for /all/ regular users at once, so it's explicitly not possible to only do it for `alice` or `bob`.

Reply:
You mean the descriptions? Because `systemd.user.tmpfiles.rules` affects all users, but `systemd.user.tmpfiles.users.$USER.rules` only `$USER`.

Reply:
Now I understood - this adds both options. I updated the PR title and suggested a small change in the "global" user tmpfile option. Maybe you can include that, update the PR title and add a description to the commit message? Otherwise LGTM.